bmu/Src/F7_Src/ltc_common.c

```diff
@@ -138,14 +162,23 @@ int batt_spi_wakeup(bool sleeping)
     // Wake up the serial interface on device S1.
     if (sleeping) {
 		HAL_GPIO_WritePin(ISO_SPI_NSS_GPIO_Port, ISO_SPI_NSS_Pin, GPIO_PIN_RESET);
-		delay_us(300);
+		/* TODO: take out of common file */
+        long_delay_us(LTC_T_WAKE_MAX_US * NUM_BOARDS); // Have to be careful here, with 14 AMS boards we wait 2.3ms and t_IDLE is 2.4ms
 		HAL_GPIO_WritePin(ISO_SPI_NSS_GPIO_Port, ISO_SPI_NSS_Pin, GPIO_PIN_SET);
-		delay_us(10);
+
+#if (LTC_T_WAKE_MAX_US * NUM_BOARDS >= T_IDLE_US) // Included as suggested by datasheet
+        if (batt_spi_tx(&dummy, JUNK_SIZE))
+        {
+            ERROR_PRINT("Failed to wake isospi after wake to core\n");
+            return 2;
```

Comment:
Should return an enum that specifies meaningful mappings for this. Makes it easier to debug